src/auctionManager.js

```diff
@@ -0,0 +1,452 @@
+        }
+        if (bid.mediaType === 'video' && !bid.vastUrl) {
+          utils.logError(errorMessage(`Video bid does not have required vastUrl property.`));
+          return false;
+        }
+        return true;
+      }
+
+      // Postprocess the bids so that all the universal properties exist, no matter which bidder they came from.
+      // This should be called before addBidToAuction().
+      function prepareBidForAuction() {
+        // Let listeners know that now is the time to adjust the bid, if they want to.
+        //
+        // This must be fired first, so that we calculate derived values from the updates
+        events.emit(CONSTANTS.EVENTS.BID_ADJUSTMENT, bid);
+
+        let bidderRequest = auction.getBidderRequests().find(request => {
+          return request.bids
+            .filter(rbid => rbid.bidder === bid.bidderCode && rbid.placementCode === adUnitCode).length > 0;
+        }) || {start: null};
+
+        const start = bidderRequest.start;
+
+        Object.assign(bid, {
+          auctionId: auctionId,
+          responseTimestamp: timestamp(),
+          requestTimestamp: start,
+          cpm: parseFloat(bid.cpm) || 0,
+          bidder: bid.bidderCode,
+          adUnitCode
+        });
+
+        bid.timeToRespond = bid.responseTimestamp - bid.requestTimestamp;
+
+        const priceStringsObj = getPriceBucketString(bid.cpm, _customPriceBucket);
+        bid.pbLg = priceStringsObj.low;
+        bid.pbMg = priceStringsObj.med;
+        bid.pbHg = priceStringsObj.high;
+        bid.pbAg = priceStringsObj.auto;
+        bid.pbDg = priceStringsObj.dense;
+        bid.pbCg = priceStringsObj.custom;
+
+        // if there is any key value pairs to map do here
+        var keyValues = {};
+        if (bid.bidderCode && (bid.cpm > 0 || bid.dealId)) {
+          keyValues = getKeyValueTargetingPairs(bid.bidderCode, bid);
+        }
+
+        bid.adserverTargeting = keyValues;
+      }
+
+      function doCallbacksIfNeeded() {
+        if (bid.timeToRespond > $$PREBID_GLOBAL$$.cbTimeout + $$PREBID_GLOBAL$$.timeoutBuffer) {
+          auction.executeCallback(true);
+        }
+      }
+
+      // Add a bid to the auction.
+      function addBidToAuction() {
+        events.emit(CONSTANTS.EVENTS.BID_RESPONSE, bid);
+        auction.setBidsReceived(bid);
+      }
+    }
+
+    this.callBids = (cbTimeout) => {
+      this.setAuctionStatus(AUCTION_STARTED);
+      this.setAuctionId(utils.generateUUID());
+      this.setAuctionStart(Date.now());
+
+      const auctionInit = {
+        timestamp: this.getAuctionStart(),
+        auctionId: this.getAuctionId(),
+        timeout: cbTimeout
+      };
+      events.emit(CONSTANTS.EVENTS.AUCTION_INIT, auctionInit);
+
+      let bidRequests = adaptermanager.makeBidRequests(this.getAdUnits(), this.getAuctionStart(), this.getAuctionId(), cbTimeout);
+      bidRequests.forEach(bidRequest => {
+        this.setBidderRequests(bidRequest);
+      });
+      let doneCb = this.done();
```

Comment:
`this.done()` _executes_ the `done` function. `this.done` _references_ the `done` function.

This will call `done()` immediately--before `callBids` even runs.